source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -180,11 +282,69 @@ void TextureBaker::writeBakedDocument(NodePtr shader, const FilePath& filename)
     writeToXmlFile(bakedTextureDoc, filename);
 }
 
-FilePath TextureBaker::generateTextureFilename(OutputPtr output)
+FilePath TextureBaker::generateTextureFilename(OutputPtr output, const string& srName, const string& udim)
 {
     string outputName = createValidName(output->getNamePath());
-    string udimSuffix = _udim.empty() ? EMPTY_STRING : "_" + _udim;
-    return FilePath(outputName + "_baked" + udimSuffix + "." + _extension);
+    string srSegment = srName.empty() ? EMPTY_STRING : "_" + srName;
+    string udimSuffix = udim.empty() ? EMPTY_STRING : "_" + udim;
+
+    return FilePath(outputName + srSegment + "_baked" + udimSuffix + "." + _extension);
+}
+
+void TextureBaker::bakeAllShaders(DocumentPtr& doc, string file, bool hdr, int texres)
+{
+    TextureBakerPtr baker = TextureBaker::create(texres, texres, hdr ? Image::BaseType::FLOAT : Image::BaseType::UINT8);
+    FileSearchPath filename = initFileSearchPath();
+    GenContext genContext = initGenContext();
+    genContext.registerSourceCodeSearchPath(filename);
+    StringVec renderablePaths = getRenderablePaths(doc);
+    StringResolverPtr resolver = StringResolver::create();
+    for (const auto& renderablePath : renderablePaths)
+    {
+        ElementPtr elem = doc->getDescendant(renderablePath);
+        TypedElementPtr typedElem = elem ? elem->asA<TypedElement>() : nullptr;
+        if (!typedElem || elem->getCategory() != "shaderref")
+        {
+            continue;
+        }
+        ShaderRefPtr sr = elem->asA<ShaderRef>();
+        FilePath out = generateOutMtlxFilename(file, sr->getName());
+        StbImageLoaderPtr stbimg = StbImageLoader::create();
+        ImageHandlerPtr imageHandler = GLTextureHandler::create(stbimg);
+
+        // Check for any udim set.
+        ValuePtr udimSetValue = doc->getGeomPropValue("udimset");
+
+        if (udimSetValue && udimSetValue->isA<StringVec>())
+        {
+            for (const string& udim : udimSetValue->asA<StringVec>())
```

Comment:
To avoid code repeat, a string array could be kept of all udim strings. If there are none then a single entry with a EMPTY_STRING could be used to pass as the udim.

Reply:
Good idea! Thanks!